### ISSUE TITLE
twigphp/twig-cache-extension is not maintained anymore

### ISSUE THREAD

**Issue by gchtr** [timber/timber#2407]
As @nlemoine [mentioned](https://github.com/timber/timber/pull/2349#issuecomment-760306256), the [twigphp/twig-cache-extension](https://github.com/twigphp/twig-cache-extension) is not maintained anymore. Instead, [twigphp/cache-extra](https://github.com/twigphp/cache-extra) should be used.

The only information I found about this in [a tweet](https://twitter.com/fabpot/status/1346406037042171911).

We should check out what this means in more detail, and maybe add the relevant changes for 1.x and 2.x. And possibly bump the supported Twig version for 2.x, because twigphp/cache-extra requires a [minimum Twig version of 2.4](https://github.com/twigphp/cache-extra/blob/3.x/composer.json#L20).

**Comment by acobster** [comment]
@gchtr looks like the current minimum Twig is ``^1.41|^2.10`` on [`master`](https://github.com/timber/timber/blob/master/composer.json#L32), but on [`2.x`](https://github.com/timber/timber/blob/2.x/composer.json#L32) it's already just `^2.10`. I agree we should look into the new extension though. Nice catch!

**Comment by gchtr** [comment]
Okay, because this is connected to #2417, I checked it out a little more.

If we use [twigphp/cache-extra](https://github.com/twigphp/cache-extra), then we’d need to change one or two things in Timber. I didn’t look into it more closely, and I can’t quite assess now what this needs in terms of breaking changes or backwards compatibility. But I think there could be quite some work involved.

**Timber\Loader::_get_cache_extension**

https://github.com/timber/timber/blob/95e6cebf31c3b238f568507bfed029e75b407158/lib/Loader.php#L599-L615

**Timber\Cache\Keygenerator**

https://github.com/timber/timber/blob/95e6cebf31c3b238f568507bfed029e75b407158/lib/Cache/KeyGenerator.php#L1-L7

**Timber\Cache\WPObjectCacheAdapter**

https://github.com/timber/timber/blob/95e6cebf31c3b238f568507bfed029e75b407158/lib/Cache/WPObjectCacheAdapter.php#L1-L6

These are all features that I don’t find in the [twigphp/cache-extra](https://github.com/twigphp/cache-extra), because that new package is about providing a `cache` tag in Twig that allows to cache template fragments.

Maybe [Symfony’s Cache](https://symfony.com/doc/current/components/cache.html) component is something that would be more suitable because the twigphp/cache-extra package uses it was well, but maybe it’s also overkill.

Maybe it’s more about supporting PSR-6 and PSR-16. There were some major rewrites for the caching mechanism proposed in https://github.com/timber/timber/pull/1500.

Maybe it’s all much simpler and we don’t need a replacement at all.

When I look at the [version requirements](https://github.com/twigphp/twig-cache-extension/blob/d1a36989964f81f87228413ef437ab13608569f4/composer.json#L14-L16) of the unmaintained [twigphp/twig-cache-extension](https://github.com/twigphp/twig-cache-extension) package, then I see that we could still use the old cache extension with our updated minimum requirements (current discussion in https://github.com/timber/timber/discussions/2436):

```json
"require": {
    "php": ">=7.2.5",
    "symfony/cache": "^5.0|^6.0",
    "twig/twig": "^2.4|^3.0"
},
```

I’m saying this because I feel like we are low on resources for this. In case anybody wants to tackle this, they’re welcome to help here.

Let’s leave this issue to see what needs to be done about getting rid of the unmaintained extension and continue the discussion about version requirements in https://github.com/timber/timber/discussions/2436.

**Comment by gchtr** [comment]
Closed via #2546.
